modules/telemetry/include/hephaestus/telemetry/measure_sink.h

```diff
@@ -0,0 +1,29 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <chrono>
+#include <string>
+
+namespace heph::telemetry {
+using ClockT = std::chrono::system_clock;
+
+struct MeasureEntry {
+  [[nodiscard]] auto operator==(const MeasureEntry& other) const -> bool = default;
+
+  std::string component;
+  std::string tag;
+  ClockT::time_point measure_timestamp;
+  std::string json_values;
+};
+
+class IMeasureSink {
+public:
+  virtual ~IMeasureSink() = default;
+
+  virtual void send(const MeasureEntry& measure_entry) = 0;
```

Comment:
nit: when the variable type is explanatory like this, I fund it good style to leave the variable name out, since its mostly duplicated info